clientv3/watch.go

```diff
@@ -314,12 +320,7 @@ func (w *watcher) Close() (err error) {
 }
 
 func (w *watchGrpcStream) Close() (err error) {
-	w.mu.Lock()
```

Comment:
The sharing was making it unnecessarily difficult to reason about the `watcherStream` teardown path so now `watcherStream` will post itself to `watchGrpcStream.closingc` and `watchGrpcStream.run` handles the final removal of `watcherStream` resources.